Sources/MongoSwift/APM.swift

```diff
@@ -1,14 +1,109 @@
 import CLibMongoC
 import Foundation
 
-/// A protocol for monitoring events to implement, specifying their name.
-public protocol MongoEvent {
-    /// The name this event will be posted under.
-    static var eventName: Notification.Name { get }
+/// A protocol for `CommandEvent` handlers to implement.
+public protocol CommandEventHandler {
+    /// Handle a `CommandEvent`.
+    func handleCommandEvent(_ event: CommandEvent)
+}
+
+/// A protocol for handlers of events relating to SDAM to implement.
+public protocol SDAMEventHandler {
+    /// Handle a `TopologyEvent`.
+    /// Implementing this method is optional. If no implementation is provided, `TopologyEvent`s will be ignored.
+    func handleTopologyEvent(_ event: TopologyEvent)
+
+    /// Handle a `ServerHeartbeatEvent`.
+    /// Implementing this method is optional. If no implementation is provided, `ServerHeartbeatEvent`s will be ignored.
+    func handleServerHeartbeatEvent(_ event: ServerHeartbeatEvent)
+}
+
+public extension SDAMEventHandler {
+    /// Handle a `TopologyEvent`.
+    /// Implementing this method is optional. If no implementation is provided, `TopologyEvent`s will be ignored.
+    func handleTopologyEvent(_: TopologyEvent) {}
+
+    /// Handle a `ServerHeartbeatEvent`.
+    /// Implementing this method is optional. If no implementation is provided, `ServerHeartbeatEvent`s will be ignored.
+    func handleServerHeartbeatEvent(_: ServerHeartbeatEvent) {}
+}
+
+/// A protocol for events that are directly consumable by users to implement.
+private protocol Publishable {
```

Comment:
The following three private protocols help maintain the previous approach of all callbacks funneling into a single function. I also reused some of the approaches from the internal cursor implementation for dealing with the mongoc events.